src/components/SearchBox/SearchBox.js

```diff
@@ -207,11 +211,29 @@ class SearchBox extends React.Component {
         // Set a dynamic value for custom dimension2
         gaConfig.customDimensions.dimension2 = GASearchedRepo;
 
-        // Send GA "Search" Catalog, "Query Sent" Action Event
-        utils.trackSearchQuerySend(searchInputValue, gaConfig.customDimensions);
+        // 3 phase to handle GA event. We need to prevent sending extra GA events after the search
+        // request is made.
+        if (isSearchRequested && !isGAResponseReceived) {
+          return false;
+        }
+
+        if (isSearchRequested && isGAResponseReceived) {
+          return true;
```

Comment:
If we return `true` here will the search submit still work? I only ask as the it looks like the SERP is loaded by running `window.location.assign(requestUrl);`, so I'm wondering if we still need to do that. Hopefully it doesn't need that for the form to work.

Reply:
We need a `return` here to determine the function I think.

Reply:
I'm just worried that without doing an explicit `window.location.assign(requestUrl);` pressing submit won't actually do a search (unfortunately we have to handle this stuff differently for each different search we have across the site... :-( )

Reply:
we can add `window.location.assign(requestUrl);` there just in case. Actually logically, I think we should and also enable another GA QuerySent on phase 3 for a different search request. But I thought we have the assumption that the patrons will be redirected to another page like they are on Drupal Search.

Reply:
Added it!!